src/non_windows.rs

```diff
@@ -1,45 +1,142 @@
-use std::process::Command;
-use std::path::PathBuf;
-use std::env;
+use std::process::{Command, Stdio};
+use std::path::{PathBuf, Path};
 
+#[derive(Debug, Clone)]
+enum ToolKind {
+    /// LLVM-rc. Note that LLVM-RC requires a separate C preprocessor to
+    /// preprocess the rc file.
+    LlvmRc { rc: String },
+    /// MinGW windres.
+    WindRes { exec: String },
+}
 
-#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Clone)]
 pub struct ResourceCompiler {
-    windres: Option<&'static str>,
+    tool: Option<ToolKind>,
 }
 
 
 impl ResourceCompiler {
     pub fn new() -> ResourceCompiler {
-        ResourceCompiler { windres: get_windres_executable() }
+        ResourceCompiler { tool: find_rc_tool() }
     }
 
     #[inline]
     pub fn is_supported(&self) -> bool {
-        self.windres.is_some()
+        match std::env::var("TARGET").as_deref() {
+            Ok("x86_64-pc-windows-msvc") => true,
+            Ok("i686-pc-windows-msvc") => true,
+            Ok("x86_64-pc-windows-gnu") => true,
+            Ok("i686-pc-windows-gnu") => true,
```

Comment:
Do you happen to know what the aarch64 windows tuples are?

Reply:
aarch64-pc-windows-msvc. AFAIK there isn't any aarch64 mingw target yet.

Also noticed that there's an additional i586-pc-windows-msvc target, it should probably be added as well.

Alternatively, we might want to avoid using target triples and instead use `CARGO_CFG_TARGET_ARCH`, `CARGO_CFG_TARGET_OS` and `CARGO_CFG_TARGET_ENV` so we don't need to pattern match the target tuple. CARGO_CFG_TARGET_ARCH will be x86 for all i386+ targets.

Reply:
Resolved this by not being strictly denominational about the quads.